src/material-experimental/mdc-radio/harness/radio-harness.ts

```diff
@@ -8,7 +8,78 @@
 
 import {ComponentHarness, HarnessPredicate} from '@angular/cdk-experimental/testing';
 import {coerceBooleanProperty} from '@angular/cdk/coercion';
-import {RadioButtonHarnessFilters} from './radio-harness-filters';
+import {RadioButtonHarnessFilters, RadioGroupHarnessFilters} from './radio-harness-filters';
+
+/**
+ * Harness for interacting with a standard mat-radio-group in tests.
+ * @dynamic
+ */
+export class MatRadioGroupHarness extends ComponentHarness {
+  static hostSelector = 'mat-radio-group';
+
+  /**
+   * Gets a `HarnessPredicate` that can be used to search for a radio-group with
+   * specific attributes.
+   * @param options Options for narrowing the search:
+   *   - `id` finds a radio-group with specific id.
+   * @return a `HarnessPredicate` configured with the given options.
+   */
+  static with(options: RadioGroupHarnessFilters = {}): HarnessPredicate<MatRadioGroupHarness> {
+    return new HarnessPredicate(MatRadioGroupHarness)
+        .addOption('id', options.id, async (harness, id) => (await harness.getId()) === id);
```

Comment:
@jelbourn We talked about `getName()` throwing if the radio-group has radio-button's with inconsistent names. This means that we probably can't provide a predicate for the `name` attribute. should we re-think to not throw?

Reply:
I think you could still find the group by name where it matches if _any_ of the radios inside have that name, and then you throw an error in the find if there's a mismatch, e.g.
```
The locator found a radio-group with name "${name}", but some radio-buttons
within the group have mismatching names, which is invalid.
```

Reply:
That sounds like a good idea. Just implemented it. Please have another look. 

Unfortunately it turned out a bit larger code-wise than expected, but should be okay in favor of providing a good API for the harness (hopefully the comments are sufficient)